addons/purchase/models/account_invoice.py

```diff
@@ -52,8 +52,10 @@ def _onchange_purchase_auto_complete(self):
         # Copy purchase lines.
         po_lines = self.purchase_id.order_line - self.line_ids.mapped('purchase_line_id')
         new_lines = self.env['account.move.line']
+        sequence = max(self.line_ids.mapped('sequence')) + 1 if self.line_ids else 10
```

Comment:
I think it's simpler to sort the `po_lines` by their `order_id` and then increment the sequence one by one for each new line.
That way, you can remove the modification done in `purchase.py`

To do that, maybe replace the `po_lines.filtered(lambda l: not l.display_type)` by:
- A sort: `for line in po_lines.sort('order_id'):`
- A condition inside the for loop to escape the line who should be filtered:
```python
if not line.display_type:
    continue
```

Reply:
Hello :wave: 
Thank you for your feedback :)
Correct me if I'm wrong but `po_lines` will only contain one `order_id` since `_onchange_purchase_auto_complete()` process only one purchase at a time https://github.com/odoo/odoo/blob/9db077319281788b1181668ec1f80fb637ddf401/addons/purchase/models/account_invoice.py#L53
So we need to have the last sequence of the invoice lie or resequence everything each time which is probably not ideal.